pointcloud/basic_octree_test.go

```diff
@@ -557,8 +557,19 @@ func TestBasicOctreePointcloudIngestion(t *testing.T) {
 
 // Test the functionalities involved with converting a pcd into a basic octree.
 func TestReadBasicOctreeFromPCD(t *testing.T) {
-	artifactPath := "pointcloud/test_short.pcd"
+	t.Run("reading from binary PCD to octree", func(t *testing.T) {
+		binaryArtifactPath := "slam/example_cartographer_outputs/viam-office-02-22-1/pointcloud/pointcloud_15.pcd"
```

Comment:
In the additional details section, I note that the actual creation is only 0.3secs. I will look into on parallelizing the iterate calls

Reply:
Note: the time for octree creation includes adding all the points

Reply:
Can you find out what is taking 15s now? I want to make sure that Iterate() isn't slower than expected.

Reply:
It is purely in the `validateBasicOctree` function. We do approx. 15x checks per node which in a balanced octree is `2*(numPoints) - 1`. In addition for each node we construct a new metadata object for comparison which involves iterating through all the points below it in the tree.

If this iterate that happens at every node is removed, the time goes down to 7 secs instead of 15 secs

Reply:
I could memoize the data here by starting at the leaves and going up to improve time but the check would still take at a minimum, 7 seconds. This function is only used for test evaluation so I can time box this change to be 2 hrs if this is desirable.

Reply:
Thanks, that makes sense. I think testing on a smaller octree is fine, and we don't need to optimize validateBasicOctree().

Reply:
@nicksanford based on the time sink being in the validate and caused not by the iterate, I don't think it's worth doing batching at present (the iterate takes approx. 0.2 sec on this large dataset and will be much smaller on a smaller dataset). Let me know if this works for you

Reply:
Test now takes 0.1 secs on the raw data from an rplidar (808 points)

Reply:
```
carto output: 180,000 points. + 15 secs
pointcloud/bun090.pcd: 30,000 points + 2.7 sec
rplidar_data: 800 points + 0.1 sec
```

@tessavitabile @nicksanford please confirm that we are alright with the 800 point pcd file instead of the 30,000 that takes 2.7 sec

Reply:
yes

Reply:
done